### ISSUE TITLE
Invalid Unit Test Annotations

### ISSUE THREAD

**Issue by okorshenko** [magento/magento2#18840]
### Preconditions (*)
1. Checkout latest 2.3-develop branch. 

### Steps to reproduce (*)
1. Find the following list of annotations in the unit tests:
```
- @expectedExceptionText
- @expectedMessage
- @message
- @exceptedExceptionMessage
- @exectedExceptionMessage
- @ExceptedExceptionMessage
- @expected
- @expecteExceptionMessage
- @true
- @assertException
- @paran
```

### Expected result (*)
1. Unit tests annotations related to exception message validation are fixed to use proper annotation: `@expectedExceptionMessage` according to [official documentation](https://phpunit.de/manual/6.5/en/appendixes.annotations.html#appendixes.annotations.expectedExceptionMessage)
2. Invalid annotations (like `@true`) removed or replaced with correct annotation depends on context

### Actual result (*)
1. Unit tests are using incorrect annotations thus exception messages are not properly validated

**Comment by magento-engcom-team** [comment]
Hi @okorshenko. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@okorshenko do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
@okorshenko Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-95946`, `MAGETWO-95947` were created

**Comment by okorshenko** [comment]
#mm18mx

**Comment by okorshenko** [comment]
Hi @okorshenko. Thank you for your report. 
The issue has been fixed in magento/magento2#18846 by @okorshenko in 2.3-develop branch
Related commit(s):
 - [0f0f64de5c2ece4a0d3ec9ed54f65df3c30284e9](https://github.com/magento/magento2/commit/0f0f64de5c2ece4a0d3ec9ed54f65df3c30284e9)

The fix will be available with the upcoming 2.3.1 release.

**Comment by magento-engcom-team** [comment]
Hi @okorshenko. Thank you for your report. 
The issue has been fixed in magento/magento2#19105 by @swnsma in 2.2-develop branch
Related commit(s):
 - [252a0ed8225d03ff36e0c4139e46d37db8ad7402](https://github.com/magento/magento2/commit/252a0ed8225d03ff36e0c4139e46d37db8ad7402)

The fix will be available with the upcoming 2.2.8 release.
